include/discord-internal.h

```diff
@@ -1135,6 +1136,24 @@ bool discord_message_commands_try_perform(
 
 /** @} DiscordInternalMessageCommands */
 
+/** @defgroup DiscordInternalCache Cache API
+ * @brief The Cache API for storage and retrieval of Discord data
+ *  @{ */
+
+struct discord_cache {
+    struct _discord_cache_data *data;
+    void (*cleanup)(struct discord *client);
+    void (*on_shard_disconnected)(struct discord *client,
+                                  const struct discord_identify *ident,
+                                  bool resumable);
+    void (*on_shard_resumed)(struct discord *client,
+                             const struct discord_identify *ident);
+    void (*on_shard_reconnected)(struct discord *client,
+                                 const struct discord_identify *ident);
+};
```

Comment:
We might want to document this. A tiny description for each field, and a `@brief` for the `struct discord_cache` will do